Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Calibration/EditorExtrinsicsCalibration.cs

```diff
@@ -0,0 +1,341 @@
+                }
+            }
+
+            if (headsetData != null)
+            {
+                if (headsetData.markers.Count != expectedNumberOfMarkers)
+                {
+                    Debug.Log("Headset has not yet detected all of the markers on the calibration board, dropping payload from headset.");
+                }
+                else
+                {
+                    var dslrTexture = CompositorWrapper.GetDSLRTexture();
+                    CalibrationDataHelper.SaveDSLRArUcoImage(dslrTexture, nextArUcoImageId);
+                    CalibrationDataHelper.SaveHeadsetData(headsetData, nextArUcoImageId);
+
+                    if (ProcessArUcoData(headsetData, dslrTexture))
+                    {
+                        CalibrationDataHelper.SaveDSLRArUcoDetectedImage(dslrTexture, nextArUcoImageId);
+                        CreateVisual(headsetData, nextArUcoImageId);
+                    }
+
+                    nextArUcoImageId++;
+                }
+
+                headsetData = null;
+            }
+
+            if (Input.GetKeyDown(KeyCode.Return))
+            {
+                Debug.Log("Starting Individual Camera Extrinsics calculations.");
+                cameraExtrinsics = calibration.CalculateIndividualArUcoExtrinsics(dslrIntrinsics, parentVisuals.Count);
+                if (cameraExtrinsics != null)
+                {
+                    foreach (var extrinsic in cameraExtrinsics)
+                    {
+                        Debug.Log($"Calculated extrinsics: {extrinsic}");
+                    }
+                    CreateExtrinsicsVisual(cameraExtrinsics);
+                }
+
+                Debug.Log("Starting the Global Camera Extrinsics calculation.");
+                globalExtrinsics = calibration.CalculateGlobalArUcoExtrinsics(dslrIntrinsics);
+                if (globalExtrinsics != null)
+                {
+                    var fileName = CalibrationDataHelper.SaveCameraExtrinsics(globalExtrinsics);
+                    Debug.Log($"Saved global extrinsics: {fileName}");
+                    Debug.Log($"Found global extrinsics: {globalExtrinsics}");
+                    var position = globalExtrinsics.ViewFromWorld.GetColumn(3);
+                    var rotation = Quaternion.LookRotation(globalExtrinsics.ViewFromWorld.GetColumn(2), globalExtrinsics.ViewFromWorld.GetColumn(1));
+                    GameObject camera = null;
+                    cameraVisualHelper.CreateOrUpdateVisual(ref camera, position, rotation);
+                    camera.name = "Global Extrinsics";
+                    GameObject hololens = null;
+                    cameraVisualHelper.CreateOrUpdateVisual(ref hololens, Vector3.zero, Quaternion.identity);
+                    hololens.name = "Global HoloLens";
+                }
+            }
+        }
+
+        private void OnDataReceived(string playerId, byte[] payload)
+        {
+            Debug.Log($"Received payload of {payload.Length} bytes");
```

Comment:
Might be good to funnel logs through a method and add a property that allows you to toggle them

Reply:
This seems like a good suggestion for the future but it is probably beyond the scope of what i would like to get in with this review.